hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -153,18 +166,52 @@ class WorkflowParams(param.Parameterized):
                      doc="When comparing CSV files during regression tests, use this value as the maximum allowed "
                          "relative difference of actual and expected results. Default: 0.0 (must match exactly)")
     regression_metrics: str = param.String(default=None, doc="A list of names of metrics to compare")
+    run_inference_only: bool = param.Boolean(False, doc="If True, run only inference and skip training after loading"
+                                                        "model weights from the specified checkpoint in "
+                                                        "`src_checkpoint` flag. If False, run training and inference.")
 
     CROSSVAL_INDEX_ARG_NAME = "crossval_index"
     CROSSVAL_COUNT_ARG_NAME = "crossval_count"
 
-    def validate(self) -> None:
-        if self.weights_url and self.local_weights_path:
-            raise ValueError("Cannot specify more than one of local_weights_path, weights_url.")
+    @property
+    def src_checkpoint_is_url(self) -> bool:
+        try:
+            result = urlparse(self.src_checkpoint)
+            return all([result.scheme, result.netloc])
+        except ValueError:
+            return False
+
+    @property
+    def src_checkpoint_is_local_file(self) -> bool:
+        return Path(self.src_checkpoint).exists()
 
+    @property
+    def src_checkpoint_is_aml_run_id(self) -> bool:
+        return re.match(r"[_\w-]*$", self.src_checkpoint.split(":")[0]) is not None
+
+    @property
+    def is_valid_src_checkpoint(self) -> bool:
+        if self.src_checkpoint:
+            return self.src_checkpoint_is_local_file or self.src_checkpoint_is_url or self.src_checkpoint_is_aml_run_id
+        return True
+
+    def validate(self) -> None:
+        if not self.is_valid_src_checkpoint:
+            raise ValueError(f"Invalid src_checkpoint: {self.src_checkpoint}. Please provide a valid URL, local file "
+                             "or azureml run id.")
         if self.crossval_count > 1:
             if not (0 <= self.crossval_index < self.crossval_count):
                 raise ValueError(f"Attribute crossval_index out of bounds (crossval_count = {self.crossval_count})")
 
+        if self.run_inference_only and not self.src_checkpoint:
+            raise ValueError("Cannot run inference without a src_checkpoint. Please specify a valid src_checkpoint."
+                             "You can either use a URL, a local file or an azureml run id. For custom checkpoint paths "
+                             "within an azureml run, (other than last.ckpt), provide a src_checkpoint in the format."
```

Comment:
This same message string exists in a few different places. It might be better to store it in a variable so it is easier to update in future